### PR TITLE
clp-package: Pass input paths to compression scheduler through database rather than file.

### PR DESCRIPTION
**Pull request by haiqi96**
# Description

The current way of argument passing uses an on-disk file and doesn't handle the case when compress.sh is on a different machine from compression scheduler. This change replace the on-disk file with a list of file paths. 

# Validation performed

Manually ran compress.sh with file and input-list arguments and confirmed no error.

### CODE REVIEW

**Comment by haiqi96** [comment]
In fact, the compress.py only use --input-list to to pass argumets to native/compress.py.

If replacing postional argument PATH to be -p is not preferred, we can consider removing PATH from native/compress.py and make input-list to be a required argument

File: components/clp-package-utils/clp_package_utils/scripts/native/compress.py [hunk 1]
```diff
@@ -161,13 +161,15 @@ def main(argv):
         "-t", "--tags", help="A comma-separated list of tags to apply to the compressed archives."
     )
     parsed_args = args_parser.parse_args(argv[1:])
+    compression_paths = parsed_args.paths
```
Comment (kirkrodrigues) [inline components/clp-package-utils/clp_package_utils/scripts/native/compress.py:164]: Can we do some renaming:
* `--input-list` -> `--path-list`
* `compression_paths` -> `paths_arg` (or if you want to be more specific, `compress_paths_arg`)
* `compression_input_list` -> `path_list_arg` (or if you want to be more specific `compress_path_list_arg`)
* `list_path` -> `paths_to_compress`
* `InputConfig.list_path` -> `InputConfig.paths_to_compress`

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]

File: components/clp-package-utils/clp_package_utils/scripts/native/compress.py [hunk 2]
```diff
@@ -185,26 +187,30 @@ def main(argv):
...
+        for path in compress_paths_arg:
+            stripped_path = path.strip()
+            if "" == stripped_path:
+                # Skip empty paths
+                continue
+            resolved_path_str = str(pathlib.Path(stripped_path).resolve())
+            paths_to_compress.append(resolved_path_str)
+    else:
+        # Read paths from the input file
+        log_paths_to_compress = pathlib.Path(compress_path_list_arg).resolve()
```
Comment (kirkrodrigues) [inline components/clp-package-utils/clp_package_utils/scripts/native/compress.py:201]: `compress_path_list_path`? `log_paths_to_compress` sounds like a list but in this case it's a file path to a list of log paths.

File: components/clp-package-utils/clp_package_utils/scripts/compress.py [hunk 3]
```diff
@@ -115,25 +115,25 @@ def main(argv):
...
         compress_cmd.append(path)
-    if parsed_args.input_list is not None:
+    if parsed_args.path_list is not None:
         # Get unused output path
         while True:
-            output_list_filename = f"{uuid.uuid4()}.txt"
-            output_list_path = clp_config.logs_directory / output_list_filename
-            if not output_list_path.exists():
+            container_path_list_filename = f"{uuid.uuid4()}.txt"
+            container_path_list = clp_config.logs_directory / container_path_list_filename
```
Comment (kirkrodrigues) [inline components/clp-package-utils/clp_package_utils/scripts/compress.py:122]: `container_path_list_path`?

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]

**Review by kirkrodrigues — APPROVED** [review]
How about this?

clp-package: Pass input paths to compression scheduler through database rather than file. (#307)

**Pull request merged**